tests/test_paper.py

```diff
@@ -0,0 +1,209 @@
+'''
+Paper-mode testing
+'''
+
+import trio
+from exceptiongroup import BaseExceptionGroup
+from typing import (
+    AsyncContextManager,
+    Literal,
+)
+
+import pytest
+import tractor
+from tractor._exceptions import ContextCancelled
+from uuid import uuid4
+from functools import partial
+
+from piker.log import get_logger
+from piker.clearing._messages import Order
+from piker.pp import (
+    open_trade_ledger,
+    open_pps,
+)
+
+log = get_logger(__name__)
+
+
+def get_fqsn(broker, symbol):
+    fqsn = f'{symbol}.{broker}'
+    return (fqsn, symbol, broker)
+
+
+oid = ''
+test_exec_mode = 'live'
+(fqsn, symbol, broker) = get_fqsn('kraken', 'xbtusdt')
+brokers = [broker]
+account = 'paper'
+
+
+async def _async_main(
+    open_test_pikerd_and_ems: AsyncContextManager,
+    action: Literal['buy', 'sell'] | None = None,
+    price: int = 30000,
+    executions: int = 1,
+    size: float = 0.01,
+    # Assert options
+    assert_entries: bool = False,
+    assert_pps: bool = False,
+    assert_zeroed_pps: bool = False,
+    assert_msg: bool = False,
+   ) -> None:
+    '''
+    Start piker, place a trade and assert data in 
+    pps stream, ledger and position table. 
+    '''
+
+    oid: str = ''
+    last_msg = {}
+
+    async with open_test_pikerd_and_ems() as (
+        services,
+        (book, trades_stream, pps, accounts, dialogs),
+    ):
+        if action:
+            for x in range(executions):
+                oid = str(uuid4())
+                order = Order(
+                    exec_mode=test_exec_mode,
+                    action=action,
+                    oid=oid,
+                    account=account,
+                    size=size,
+                    symbol=fqsn,
+                    price=price,
+                    brokers=brokers,
+                )
+                # This is actually a syncronous call to push a message
+                # to the async ems clue - hence why we call trio.sleep afterwards
+                book.send(order)
+
+                async for msg in trades_stream:
+                    last_msg = msg
+                    match msg:
+                        # Wait for position message before moving on
+                        case {'name': 'position'}:
+                            break
+
+        if assert_entries or assert_pps or assert_zeroed_pps or assert_msg:
```

Comment:
when you have a big logic expression like this it's much nicer for readability to break it into multi-line tuple style:

```python
               if (
            assert_entries
            or assert_pps
            or assert_zeroed_pps
            or assert_msg
        ):
```